application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/TrialLicenseGenerator.java

```diff
@@ -238,7 +238,7 @@ private URL getLicensesUpdateURL() throws URISyntaxException, MalformedURLExcept
     private Boolean isMandatoryLicensedExtension(ExtensionId extensionId)
     {
         return licensedExtensionManager.getMandatoryLicensedExtensions().stream()
-            .filter(extension -> extension.getId().contentEquals(extensionId.getId())).findFirst().isPresent();
+            .anyMatch(mandatoryExtensionId -> mandatoryExtensionId.equals(extensionId));
```

Comment:
Couldn't you simply use ``contains``?

    return licensedExtensionManager.getMandatoryLicensedExtensions().contains(extensionId);